myriad-scheduler/src/main/java/com/ebay/myriad/scheduler/event/handlers/ResourceOffersEventHandler.java

```diff
@@ -162,30 +164,31 @@ private boolean matches(Offer offer, NMProfile profile) {
     checkResource(mem < 0, "mem");
     checkResource(ports < 0, "port");
 
-    return checkAggregates(offer, profile, ports, cpus, mem);
+    return checkAggregates(offer, taskToLaunch, ports, cpus, mem);
   }
 
+    private boolean checkAggregates(Offer offer, NodeTask taskToLaunch, int ports, double cpus, double mem) {
+        Map<String, String> requestAttributes = new HashMap<>();
+        final double aggrCpu = taskUtils.getAuxTaskCpus(taskToLaunch.getProfile(), taskToLaunch.getTaskPrefix());
+        final double aggrMem = taskUtils.getAuxTaskMemory(taskToLaunch.getProfile(), taskToLaunch.getTaskPrefix());
+        if (aggrCpu <= cpus
+                && aggrMem <= mem
+                && SchedulerUtils.isMatchSlaveAttributes(offer, requestAttributes)
+                && NMPorts.expectedNumPorts() <= ports) {
+            return true;
```

Comment:
I think you should be more careful here.  We're using NMPorts which is for the NM to check ports for JHS and TMS, what if that changes?  Maybe we can get these from the profile?

Also, it seems like you'd like 3 specific ports for the JHS, maybe placing the logic here we could reject a number of offers we don't want (using them for FGS or NM's) before accepting non optimal ports.  Would like to understand what functionality we loose by accepting non optimal ports.  Can we still access all JHS pages from YARN UI?  Do jobs log their history?  If not I think we need only accept the specified ports or use HAProxy or another service discovery solution.